src/java.net.http/share/classes/jdk/internal/net/http/websocket/Frame.java

```diff
@@ -132,18 +133,20 @@ Masker mask(int value) {
          * The source and the destination buffers may be the same instance. If
          * the mask hasn't been previously set it is assumed to be 0.
          */
-        Masker transferMasking(ByteBuffer src, ByteBuffer dst) {
-            begin(src, dst);
-            loop(src, dst);
-            end(src, dst);
-            return this;
+        public void applyMask(ByteBuffer src, ByteBuffer dst) {
+            if (src.order() == dst.order()) {
+                initVectorMask(src, dst);
+                applyVectorMask(src, dst);
+            }
+            applyPlainMask(src, dst);
         }
 
-        /*
-         * Applies up to 3 remaining from the previous pass bytes of the mask.
+        /**
+         * Positions the {@link #offset} at 0, which is needed for vectorized masking, by masking necessary amount of bytes.
```

Comment:
```suggestion
         * Applies up to 3 remaining from the previous pass bytes of the mask and
         * positions the {@link #offset} at 0, which is needed for vectorized masking, 
         * by masking necessary amount of bytes.
```

Reply:
@Michael-Mc-Mahon, I'm fine with the change, though doesn't this obsolete the `by masking necessary amount of bytes` statement at the end?

Reply:
Sure, feel free to alter the suggestion. I really just wanted the original point about "up to 3 remaining from the previous pass" bit retained.

Reply:
Fixed in 6d35f0b593e097a422b192be31deebed180c8290.